// together-ai/resources/fine-tune.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
export class FineTuneResource extends APIResource {
    /**
     * Create a fine-tuning job
     */
    create(body, options) {
        return this._client.post('/fine-tunes', { body, ...options });
    }
    /**
     * Retrieve fine-tune job details
     */
    retrieve(id, options) {
        return this._client.get(`/fine-tunes/${id}`, options);
    }
    /**
     * List fine-tune job history
     */
    list(options) {
        return this._client.get('/fine-tunes', options);
    }
    /**
     * Cancels a running fine-tuning job.
     */
    cancel(id, options) {
        return this._client.post(`/fine-tunes/${id}/cancel`, options);
    }
    /**
     * Downloads a compressed fine-tuned model or checkpoint to local disk.
     */
    download(query, options) {
        return this._client.get('/finetune/download', { query, ...options });
    }
    /**
     * List events of a fine-tune job
     */
    listEvents(id, options) {
        return this._client.get(`/fine-tunes/${id}/events`, options);
    }
}
(function (FineTuneResource) {
})(FineTuneResource || (FineTuneResource = {}));
